File: rb/features.rb
# PublicHoliday SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module PublicHolidayFeatures
  def self.make_feature(name)
    case name
    when "base"
      PublicHolidayBaseFeature.new
    when "ratelimit"
      PublicHolidayRatelimitFeature.new
    when "retry"
      PublicHolidayRetryFeature.new
    when "test"
      PublicHolidayTestFeature.new
    when "timeout"
      PublicHolidayTimeoutFeature.new
    else
      PublicHolidayBaseFeature.new
    end
  end
end

File: py/features.py
# PublicHoliday SDK feature factory

from feature.base_feature import PublicHolidayBaseFeature
from feature.test_feature import PublicHolidayTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PublicHolidayBaseFeature(),
        "test": lambda: PublicHolidayTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
